debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSchema.java

```diff
@@ -344,14 +347,23 @@ protected void refreshSchemas() {
      */
     public boolean applyDdl(SourceInfo source, String databaseName, String ddlStatements,
                             DatabaseStatementStringConsumer statementConsumer) {
+        ParsingException throwException = null;
         if (ignoredQueryStatements.contains(ddlStatements)) return false;
         try {
             this.ddlChanges.reset();
             this.ddlParser.setCurrentSchema(databaseName);
             this.ddlParser.parse(ddlStatements, tables);
         } catch (ParsingException e) {
-            logger.error("Error parsing DDL statement and updating tables: {}", ddlStatements, e);
+            if (!skipUnparseableDDL) {
+                logger.error("Error parsing DDL statement and updating tables: {}", ddlStatements);
+                throwException = e;
+            } else {
+                logger.warn("Ignoring unparseable DDL statement '{}': {}", ddlStatements);
+            }
```

Comment:
Maybe reverse the if block to avoid the negation, as was done in `AbstractDatabaseHistory`?